sidekick_core/lib/sidekick_core.dart

```diff
@@ -128,11 +136,14 @@ class SidekickCommandRunner<T> extends CommandRunner<T> {
   /// and restore the previous globals
   Unmount mount() {
     final SidekickCommandRunner? oldRunner = _activeRunner;
+    final SidekickContextCache oldCache = internalSidekickContextCache;
     _activeRunner = this;
+    internalSidekickContextCache = SidekickContextCache();
     _entryWorkingDirectory = workingDirectory;
 
     return () {
       _activeRunner = oldRunner;
+      internalSidekickContextCache = oldCache;
```

Comment:
assert that `internalSidekickContextCache` is what we think it is. Don't revert when someone else wrote something into it